metrics/web/docs/features/searchable_dropdown/01_searchable_dropdown_design.md

```diff
@@ -0,0 +1,208 @@
+
+Here is a table of main `SelectionMenu` components we should use to implement the `SearchableDropdown` widget:
+
+Component | Description | Link
+----------| ----------- | -----
+`TriggerComponent` | Defines a trigger that opens the dropdown. | [API documentation](https://pub.dev/documentation/selection_menu/latest/components_configurations/TriggerComponent-class.html)
+`MenuPositionAndSizeComponent`| Defines dropdown's position, size, and constraints. | [API documentation](https://pub.dev/documentation/selection_menu/latest/components_configurations/MenuPositionAndSizeComponent-class.html)
+`AnimationComponent` | Defines a container for the dropdown. | [API documentation](https://pub.dev/documentation/selection_menu/latest/components_configurations/AnimationComponent-class.html)
+`ListViewComponent` | Defines a scrollable list of items within the dropdown. | [API documentation](https://pub.dev/documentation/selection_menu/latest/components_configurations/ListViewComponent-class.html)
+
+### Base Widget
+
+Now, let's implement the `base` `SearchableDropdown` widget. Here is a [document](https://github.com/platform-platform/monorepo/blob/master/metrics/web/docs/03_widget_structure_organization.md) that describes widget implementation in the Metrics Web Application.
+
+Consider the following class diagram that describes the `SearchableDropdown` widget structure
+and how it interacts with the `selection_menu` package.  
+![Widget class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/searchable_dropdown_document/metrics/web/docs/features/searchable_dropdown/diagrams/searchable_dropdown_widget_class_diagram.puml)
+
+Consider the following table that describes each `SearchableDropdown` field:
+
+Parameter | Type | Description                                                          
+--------- | ---- | -----------
+`searchBarBuilder` | `SearchBarBuilder` | A builder for a search bar.                  
+`dropdownMenuBuilder` | `DropdownMenuBuilder` | A builder for a dropdown menu that is a container for items to display.                 
+`itemBuilder` | `DropdownItemBuilder` | A builder for an item within the `Dropdown` that represents a single search match. 
+`dropdownMenuController` | `SelectionMenuController` | A controller that allows triggering the dropdown's menu to open or close.
+`onDropdownMenuStateChanged` | `ValueChanged<MenuState>` | A callback that is called when the dropdown menu state changed, e.g. the dropdown menu closes or opens.
+`items` | `List<T>` | A list of items to display. 
+`maxVisibleItems` | `int` | A number of visible search items.
```

Comment:
```suggestion
`maxVisibleItems` | `int` | A number of visible search items.
```
Should we follow the same approach that we have in a current dropdown element in displaying 5.5 items? Maybe this should be just height which would be more flexible then? Let me know your thoughts